vector/src/main/java/im/vector/app/features/media/VectorAttachmentViewerViewModel.kt

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2020 New Vector Ltd
+ * Copyright (C) 2018 stfalcon.com
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.features.media
+
+import com.airbnb.mvrx.MavericksViewModelFactory
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedFactory
+import dagger.assisted.AssistedInject
+import im.vector.app.core.di.MavericksAssistedViewModelFactory
+import im.vector.app.core.di.hiltMavericksViewModelFactory
+import im.vector.app.core.platform.VectorDummyViewState
+import im.vector.app.core.platform.VectorViewModel
+import im.vector.app.features.media.domain.usecase.DownloadMediaUseCase
+import im.vector.app.features.session.coroutineScope
+import kotlinx.coroutines.launch
+import org.matrix.android.sdk.api.session.Session
+
+class VectorAttachmentViewerViewModel @AssistedInject constructor(
+        @Assisted initialState: VectorDummyViewState,
+        private val session: Session,
+        private val downloadMediaUseCase: DownloadMediaUseCase
+) : VectorViewModel<VectorDummyViewState, VectorAttachmentViewerAction, VectorAttachmentViewerViewEvents>(initialState) {
+
+    /* ==========================================================================================
```

Comment:
are these headers necessary?  I would lean into them being a symptom of classes becoming too large, it also means method ordering becomes fractured rather than by usage

```kotlin
fun foo() {
  bar()
}

private fun bar() {}

fun baz() {
  fuzz()
}

private fun fuzz() {}
```

I'm not aware of project conventions for either though :sweat_smile:

Reply:
I would say it is not "necessary", but I think it make the code more structured and unified. Every class is structured the same way so we know where to look for something. But this is the way I am used to work 😄 . Maybe we can decide together of best practices to structure code.

For the size of classes, I have never had such problem of size, it adds extra space in the class for sure, but if we respect the single responsability principle for each component, in most cases there is no problem.

Good point for the order by usage, I have never thought this way.

Reply:
> For the size of classes, I have never had such problem of size, it adds extra space in the class for sure, but if we respect the single responsability principle for each component, in most cases there is no problem.

that's my thinking, if the classes are small enough, they might not be needed :crossed_fingers:

Reply:
As discussed, I removed all added section bloc comments.